packages/dht/src/connection/webrtc/BrowserWebrtcConnection.ts

```diff
@@ -19,6 +19,61 @@ interface Params {
     iceServers?: IceServer[]
 }
 
+const MESSAGE_QUEUE_MAX_SIZE = 10000
+
+class MessageQueue {
+    
+    private readonly queue: Array<Uint8Array>
+    private readonly sendFn: (message: Uint8Array) => void
+    private running: boolean
+
+    constructor(sendFn: (message: Uint8Array) => void) {
+        this.running = false
+        this.queue = []
+        this.sendFn = sendFn
+    }
+
+    push(message: Uint8Array): void {
+        if (this.queue.length >= MESSAGE_QUEUE_MAX_SIZE) {
+            logger.warn('Dropping message due to buffer overflow')
+            throw new Error('Cannot add to buffer queue full')
+        }
+        this.queue.push(message)
+    }
+
+    sendBuffer(): void {
+        if (this.running === true) {
+            return
+        }
+        this.running = true
+        let sendAttempts = 0;
+        (async () => {
+            while (this.queue.length > 0 && this.running) {
+                const messageToSend = this.queue.shift()!
+                try {
+                    this.sendFn(messageToSend)
+                } catch (err) {
+                    this.queue.unshift(messageToSend)
+                }
+                sendAttempts++
+                if (sendAttempts % 200 === 0) {
+                    await wait(0) // give up CPU cycle 
```

Comment:
I think a shorter pause should be used here see https://stackoverflow.com/questions/18826570/settimeout0-vs-window-postmessage-vs-messageport-postmessage on how to implement one. (I have tried this in the past, and the send buffer does get drained also during the shorter pauses)